src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +306,72 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var parser = new StringParser(statusFileContents, '\n');
+            var results = default(ParsedStatus);
+
+            while (parser.MoveNext())
+            {
+                string[] elements = parser.ExtractCurrent().Split(Delimiters, 3, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
We should really avoid both this ExtractCurrent and Split call.  The former is going to allocate a string, then we're going to split it which will allocate an array and multiple strings, and we don't actually need any of those as we can just parse it all with spans.

Reply:
Agreed.  I think I will have to wait for #26528 to be implemented and only after that I believe I can implement this using spans.

Reply:
FWIW I think you can do it without that.  Those APIs are just effectively helpers that wrap IndexOf.  What is the format of the line being parsed here?   Can you share an example?

Reply:
Oh, you mean a custom parser ?
Here's an example of a process status file:
```
Name:   bash
Umask:  0022
State:  S (sleeping)
Tgid:   17248
Ngid:   0
Pid:    17248
PPid:   17200
TracerPid:      0
Uid:    1000    1000    1000    1000
Gid:    100     100     100     100
FDSize: 256
Groups: 16 33 100
NStgid: 17248
NSpid:  17248
NSpgid: 17248
NSsid:  17200
VmPeak:     131168 kB
VmSize:     131168 kB
VmLck:           0 kB
VmPin:           0 kB
VmHWM:       13484 kB
VmRSS:       13484 kB
RssAnon:     10264 kB
RssFile:      3220 kB
RssShmem:        0 kB
VmData:      10332 kB
VmStk:         136 kB
VmExe:         992 kB
VmLib:        2104 kB
VmPTE:          76 kB
VmPMD:          12 kB
VmSwap:          0 kB
HugetlbPages:          0 kB        # 4.4
CoreDumping:   0                       # 4.15
Threads:        1
SigQ:   0/3067
SigPnd: 0000000000000000
ShdPnd: 0000000000000000
SigBlk: 0000000000010000
SigIgn: 0000000000384004
SigCgt: 000000004b813efb
CapInh: 0000000000000000
CapPrm: 0000000000000000
CapEff: 0000000000000000
CapBnd: ffffffffffffffff
CapAmb:   0000000000000000
NoNewPrivs:     0
Seccomp:        0
Speculation_Store_Bypass:       vulnerable
Cpus_allowed:   00000001
Cpus_allowed_list:      0
Mems_allowed:   1
Mems_allowed_list:      0
voluntary_ctxt_switches:        150
nonvoluntary_ctxt_switches:     545
```

Reply:
It may be best to get a "simple, safe" implementation into master now, so we can backport it today. Then follow up with optimizations in master.

Reply:
@danmosemsft, this code is used on existing code paths.  It is going to make these code paths slower than they are today.  You want to rush that in for 3.1?

Reply:
> Oh, you mean a custom parser ?

Yup.

Reply:
> this code is used on existing code paths. It is going to make these code paths slower than they are today. You want to rush that in for 3.1? 

No

Reply:
> To be clear I'm not just talking about these allocations; we're opening and reading from a second file now for every process.

I see, this indeed is a major concern. Just to update, I am trying to derive both values present `/proc/[pid]/stat` and `/proc/[pid]/status` from a single place probably from `/status` as most people claim `/status` to be a superset of `/stat`. Ideally should update on this by this weekend.

Reply:
> > To be clear I'm not just talking about these allocations; we're opening and reading from a second file now for every process.
> 
> @stephentoub
> I see, this indeed is a major concern. Just to update, I am trying to derive both values present `/proc/[pid]/stat` and `/proc/[pid]/status` from a single place probably from `/status` as most people claim `/status` to be a superset of `/stat`. Ideally should update on this by this weekend.

Unfortunately, its not a total superset, ***most*** of the values are there but not ***all***. So need to read both files.